### ISSUE TITLE
[Performance] Diffusion SignRound tuning is prohibitively slow for large video models

### ISSUE THREAD

**Issue by changwangss** [intel/auto-round#2279]
## Description

Diffusion calibration can now reduce denoising work by sampling a short set of scheduler timesteps, but the subsequent SignRound tuning stage remains prohibitively slow for large video diffusion models such as WAN 2.2.

This is not caused by rerunning the diffusion scheduler during tuning. The bottleneck is the per-block optimization loop over very large cached video activations.

## Reproduction

Model:

```text
/data3/changwa1/Wan2.2-T2V-A14B-Diffusers
```

Command (one physical A100 80 GB, exposed as logical device 0):

```bash
CUDA_VISIBLE_DEVICES=3 auto-round \
  --model /data3/changwa1/Wan2.2-T2V-A14B-Diffusers \
  --scheme NVFP4 \
  --format llm_compressor \
  --device_map 0 \
  --batch_size 8 \
  --nsamples 64 \
  --iters 200 \
  --calib_num_inference_steps 8 \
  --low_gpu_mem_usage \
  --output_dir /mnt/disk3/changwa1/Wan2.2-T2V-A14B-Diffusers-NVFP4 \
  --model_dtype bf16
```

Environment:

- GPU: NVIDIA A100-SXM4-80GB
- WAN pipeline defaults: 81 frames, 480x832
- AutoRound source install

## Observed performance

Calibration:

```text
cache block inputs: 64/64 [4:34:02, 256.92s/it]
peak_ram: 200.36 GB
peak_vram: 0.67 GB
```

Tuning:

```text
blocks.0: ~55m 07s
blocks.1: ~56m 56s
blocks.2: ~56m 37s
blocks.3: ~55m 29s
blocks.4: ~55m 39s
blocks.5: ~55m 20s
blocks.6: ~55m 05s
blocks.7: ~55m
peak_ram: 440.2 GB
peak_vram: 72.58 GB
```

There are 40 blocks per WAN transformer, and WAN 2.2 has `transformer` plus `transformer_2`. At roughly 55 minutes per block, a complete run is potentially multiple days.

GPU sampling during tuning showed bursty utilization rather than sustained compute, with SM utilization fluctuating between roughly 9% and 81% and intermittent large PCIe transfers. This is consistent with repeated CPU cache selection/concatenation and CPU-to-GPU transfer under `low_gpu_mem_usage`.

## Likely cause

For diffusion with `iters > 0`, `DiffusionMixin` converts:

```text
batch_size=8, gradient_accumulate_steps=1
```

into:

```text
micro batch_size=1, gradient_accumulate_steps=8
```

Therefore, each block performs approximately:

```text
200 optimizer iterations * 8 serial micro-batches
= 1600 block forward/backward passes
```

The effective batch is preserved, but it receives no batch-parallel throughput benefit.

WAN's default video activations are also very large. The approximate token count is:

```text
21 temporal latent positions * 30 * 52 spatial patches = 32,760 tokens
```

With hidden size 5,120, one BF16 hidden-state sample alone is around 320 MiB before attention, FFN, reference output, backward graph, and QDQ intermediates.

With `low_gpu_mem_usage`, cached inputs and FP outputs remain on CPU. Each tuning micro-batch repeatedly selects and concatenates cached tensors and transfers them to the accelerator. The current job used more than 440 GB peak host RAM while showing only bursty GPU utilization.

## Expected behavior

Diffusion tuning should provide a practical way to retain an effective batch size such as 8 without serializing all eight samples on one GPU. Runtime should scale reasonably with `iters`, number of blocks, and available GPUs.

### 1. Separate batch concepts

Introduce explicit settings such as:

- `calib_batch_size`
- `tuning_micro_batch_size`
- `effective_batch_size`

The current `batch_size` silently changes meaning for diffusion.

### 2. Adaptive tuning micro-batch

Probe `8 -> 4 -> 2 -> 1` on the first block, catch CUDA OOM, and adjust gradient accumulation to preserve the requested effective batch size.

For WAN at the default resolution, the selected micro-batch will likely still be 1, but smaller diffusion models could use true batching automatically.

### 3. Diffusion data parallel tuning

Distribute cached samples across GPUs, replicate only the current block, all-reduce rounding/min-max gradients, and perform one optimizer update for the effective batch. For example:

```text
4 GPUs * micro_batch_per_gpu 1 * grad_accumulate 2 = effective batch 8
```

This is preferable to the existing LLM-style block sharding because a single WAN sample already nearly fills one 80 GB GPU.

### 4. Optimize CPU cache transfers

- Pinned host-memory cache
- Non-blocking H2D copies
- Prefetch/double buffering for the next micro-batch
- Avoid repeated Python-list `torch.cat` allocation where possible
- Store cache in contiguous or shape-bucketed tensors

### 5. Allow smaller calibration video shapes

Ensure `pipeline_call_kwargs` is honored by diffusion calibration so users can set `num_frames`, `height`, and `width`. This reduces cached activation size and tuning cost substantially.

### 6. Add profiling and progress details

Report per block:

- effective batch size
- actual micro-batch size
- gradient accumulation steps
- number and size of cached inputs
- FP-reference time versus optimization-loop time
- H2D transfer time versus block compute time

## Acceptance criteria

- Diffusion batch semantics are explicit rather than silently forcing `batch_size=1`.
- Smaller diffusion models can use a true tuning micro-batch greater than 1 when memory permits.
- Large models can use multi-GPU data parallel tuning or another parallel mechanism.
- A WAN 2.2 benchmark shows materially improved GPU utilization and per-block runtime without changing the effective optimization batch or quantization result beyond expected numerical tolerance.
